mixnet/src/fragment.rs

```diff
@@ -1,22 +1,249 @@
+use std::{collections::HashMap, ops::Deref};
+
+use sphinx_packet::{constants::PAYLOAD_SIZE, payload::PAYLOAD_OVERHEAD_SIZE};
+use uuid::Uuid;
+
 use crate::error::MixnetError;
 
+pub(crate) struct FragmentSet(Vec<Fragment>);
+
+const MAX_PLAIN_PAYLOAD_SIZE: usize = PAYLOAD_SIZE - PAYLOAD_OVERHEAD_SIZE;
+
+impl FragmentSet {
+    pub(crate) fn new(msg: &[u8]) -> Result<Self, MixnetError> {
+        let chunk_size = MAX_PLAIN_PAYLOAD_SIZE - FRAGMENT_HEADER_SIZE;
```

Comment:
NIT: `chunk_size` here can also be `const`

Reply:
Thanks. Fixed 9136f2a